polaris-react/.storybook/GridOverlay/GridOverlay.tsx

```diff
@@ -28,6 +35,7 @@ export function GridOverlay({inFrame, maxWidth, layer, children}: Props) {
   const className = classNames('GridOverlay', inFrame && 'inFrame');
   const style = {
     maxWidth,
+    margin: componentLayout === 'padded' ? '0 1rem' : '',
```

Comment:
What if rather than a having to a prop on the component to control the margin, the styles set a margin detect based on the presence of the `sb-main-padded` / `sb-main-fullscreen` / `sb-main-centered` classes that get set on the body element depending on the layout in use.

```css
.sb-main-padded .GridOverlay { margin: 0 1rem; }
```

 That way you'd only need to configure the layout of the component, rather than the layout and and then this componentLayout prop to make the grid match.